pkg/apply/merge.go

```diff
@@ -6,11 +6,21 @@ import (
 	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+func MergeMetadataForUpdate(current, updated *uns.Unstructured) error {
```

Comment:
Can you add a simple docblock explaining what this does, and why we do it?

Separately, if you want to, you could move the mergeAnnotations and mergeLabels in to this function, and just call it last in MergeObjectForUpdate. Up to you.